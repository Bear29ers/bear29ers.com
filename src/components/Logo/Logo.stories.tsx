import Logo from './Logo';

import type { Meta, StoryObj } from '@storybook/react';

const meta: Meta<typeof Logo> = {
  component: Logo,
};

export default meta;

type Story = StoryObj<typeof Logo>;

export const Standard: Story = {
  render: () => <Logo />,
};
