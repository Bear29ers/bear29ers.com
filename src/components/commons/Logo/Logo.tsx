'use client';

import type { FC } from 'react';

import { motion } from 'framer-motion';
import Link from 'next/link';

const MotionLink = motion(Link);

const Logo: FC = () => {
  return (
    <motion.div className="flex-center mt-2">
      <MotionLink
        href="/"
        className="flex-center h-16 w-16 rounded-full bg-dark text-2xl font-bold text-light"
        whileHover={{
          backgroundColor: [
            '#121212',
            'rgba(131,58,180,1)',
            'rgba(253,29,29,1)',
            'rgba(252,176,69,1)',
            'rgba(131,58,180,1)',
            '#121212',
          ],
          transition: { duration: 1, repeat: Infinity },
        }}
        role="link">
        BR
      </MotionLink>
    </motion.div>
  );
};

export default Logo;
