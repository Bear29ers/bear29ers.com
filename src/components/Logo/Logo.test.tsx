import { render } from '@testing-library/react';

import Logo from './Logo';

describe('src/components/Logo/Logo.test.tsx', () => {
  it('should be [role=link] with name BR', () => {
    const { getByRole } = render(<Logo />);
    expect(getByRole('link', { name: 'BR' })).toBeInTheDocument();
  });
});
