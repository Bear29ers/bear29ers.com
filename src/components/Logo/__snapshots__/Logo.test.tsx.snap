// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/Logo/Logo.test.tsx renders correctly 1`] = `
<div>
  <div
    class="flex-center mt-2"
  >
    <a
      class="flex-center h-16 w-16 rounded-full bg-dark text-2xl font-bold text-light"
      href="/"
    >
      BR
    </a>
  </div>
</div>
`;
